### PR TITLE
wire velocity corrections and plane shifts

### PR DESCRIPTION
**Pull request by hszumila**
This contains the wire velocity corrections for the new shms drift chambers. If the geometry parameters that specify the new shms chamber readout are not specified in the param file, then this defaults to the old behavior. No changes are observed really in the width of tracking residuals. Also, the hit positions of the center of each wire can now take in the per plane shifts from alignment.

### CODE REVIEW

File: src/THcDC.cxx [hunk 1]
```diff
@@ -104,7 +108,7 @@ void THcDC::Setup(const char* name, const char* description)
   } else {
     fHMSStyleChambers = 0;
   }
-
+  cout<<"HMS Style??\t"<<fHMSStyleChambers<<endl;
```
Comment (vargasa) [inline src/THcDC.cxx]: We can use `TError::Info() / TError::Error() ` to keep output organized ;)

**Comment by MarkKJones** [comment]
I would suggest that Xpos and YPos just be the offsets and not xcenter+offset. This will make better
compatibility with the 6 geV data and HMS in the KPP run. Also then XPos and YPos can be "optional" in the DB_request call. The Get_Center method in THcDC would have to have arguments of
Double_t GetCenter(Int_t ncham, Int_t plane) 
Then rotate XCenter+Xpos and YCenter+YPos

The XPos,YPos, fReadoutLR,fReadoutTB should be initialized to zero in ReadDataBase since they are optional parameters. 

 GetReadoutSide seems very specific to the SHMS chambers. Are the new HMS
chambers going to be the same?

**Comment by MarkKJones** [comment]
I noticed in src/THcDriftChamber.h the variables fXPos and fYPos are not used (fXCenter and fycenter wer 
not used previously). Just the variable fCenter is used. So fXpos and fYPos can be eliminated just to aviod
confusion.

File: src/THcDC.cxx (hunk 1 above)
Comment (hszumila) [reply src/THcDC.cxx]: I missed this during my de-bugging phase. I suppressed this in the update.

**Comment by hszumila** [comment]
I actually had been testing a version of the code with the optional database call in a separate branch. I've updated it here, but I can still try to eliminate the fxpos and fypos. I didn't see any other variable for setting the individual plane positions. For now, these variables default to the chamber x and y position. If there are additional offsets per plane, then it can read these in (optional) and set those positions.

**Comment by hszumila** [comment]
GetReadoutSide is very specific to the SHMS chambers. I'll take a look at the new HMS chambers and see how it is different. This function may need to be modified to be used for the chambers in both arms.

**Comment by MarkKJones** [comment]
Having fXpos and fYPos arrays in THcDC.h is fine. But in THcDriftChamberPlane.cxx fCenter is filled
using the fParent->GetCenter() and the scaler variables fXPos and fYPos declared in THcDriftChamberPlane.h are not used.

**Comment by hszumila** [comment]
I had modified GetCenter n line 73 of THcDC.h. as follows:
Double_t GetCenter(Int_t plane) const {
    return
      //fXCenter[chamber]*sin(fAlphaAngle[plane-1]) +
      //fYCenter[chamber]*cos(fAlphaAngle[plane-1]);

      fXPos[plane-1]*sin(fAlphaAngle[plane-1]) +
      fYPos[plane-1]*cos(fAlphaAngle[plane-1]);
  }

**Comment by hszumila** [comment]
I re-ran some checks on how the database reads in the x,y positions of the planes and defaults to the chamber x,y positions. Those changes are now in this PR.

**Comment by MarkKJones** [comment]
I pulled down the request. I could not figure out a clean way in git to use it directly. I used the files and made a separate pull request.

**Pull request closed without merging**